pandas/io/excel/_openpyxl.py

```diff
@@ -440,7 +442,6 @@ def write_cells(
                     target_index = self.book.index(old_wks)
                     del self.book[sheet_name]
                     wks = self.book.create_sheet(sheet_name, target_index)
-                    self.sheets[sheet_name] = wks
```

Comment:
Why can we remove this? Does create_sheet modify self.book?

Reply:
That's correct

```
from openpyxl import load_workbook

book = load_workbook("test.xlsx")
print({name: book[name] for name in book.sheetnames})
book.create_sheet("test_sheet", 0)
print({name: book[name] for name in book.sheetnames})
```

gives

```
{'Sheet1': <Worksheet "Sheet1">}
{'test_sheet': <Worksheet "test_sheet">, 'Sheet1': <Worksheet "Sheet1">}
```

Reply:
Thx